### PR TITLE
kafka sender: react to kafka send errors

### PR DESCRIPTION
**Pull request by mts-aiven**
Errors in Kafka send() results were silently ignored. Python kafka library returns future, which may (eventually) contain error related to send.

### CODE REVIEW

**Comment by mts-aiven** [comment]
Shutting down kafka while journalpump running now triggers error and reconfig through kafka send() future:

```
2020-12-23 10:00:42,312	LogSender:host1	ERROR	Unexpected exception during send to kafka
Traceback (most recent call last):
  File "/home/mts/git/journalpump/journalpump/senders/kafka.py", line 120, in send_messages
    record_metadata = fut_record_metadata.get(timeout=1)
  File "/home/mts/.local/lib/python3.9/site-packages/kafka/producer/future.py", line 65, in get
    raise self.exception # pylint: disable-msg=raising-bad-type
kafka.errors.KafkaTimeoutError: KafkaTimeoutError: Batch for TopicPartition(topic='journals', partition=0) containing 12 record(s) expired: 30 seconds have passed since last append
2020-12-23 10:00:42,312	LogSender:host1	INFO	Sleeping for 1 seconds
2020-12-23 10:00:42,825	LogSender:host1	INFO	Initializing Kafka client, address: '127.0.0.1:9092'
2020-12-23 10:00:43,827	LogSender:host1	ERROR	Problem sending 163957 messages
Traceback (most recent call last):
  File "/home/mts/git/journalpump/journalpump/senders/kafka.py", line 120, in send_messages
    record_metadata = fut_record_metadata.get(timeout=1)
  File "/home/mts/.local/lib/python3.9/site-packages/kafka/producer/future.py", line 65, in get
    raise self.exception # pylint: disable-msg=raising-bad-type
kafka.errors.KafkaTimeoutError: KafkaTimeoutError: Batch for TopicPartition(topic='journals', partition=0) containing 12 record(s) expired: 30 seconds have passed since last append

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mts/git/journalpump/journalpump/senders/base.py", line 226, in get_and_send_messages
    if self.send_messages(messages=message_bodies, cursor=messages_batch[-1][1]):
  File "/home/mts/git/journalpump/journalpump/senders/kafka.py", line 134, in send_messages
    self._init_kafka()
  File "/home/mts/git/journalpump/journalpump/senders/kafka.py", line 84, in _init_kafka
    kafka_producer = KafkaProducer(**producer_config)
  File "/home/mts/.local/lib/python3.9/site-packages/kafka/producer/kafka.py", line 381, in __init__
    client = KafkaClient(metrics=self._metrics, metric_group_prefix='producer',
  File "/home/mts/.local/lib/python3.9/site-packages/kafka/client_async.py", line 244, in __init__
    self.config['api_version'] = self.check_version(timeout=check_timeout)
  File "/home/mts/.local/lib/python3.9/site-packages/kafka/client_async.py", line 900, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable
```

File: journalpump/senders/kafka.py [hunk 1]
```diff
@@ -110,9 +110,14 @@ def send_messages(self, *, messages, cursor):
         if not self.kafka_producer:
             self._init_kafka()
         try:
-            for msg in messages:
-                self.kafka_producer.send(topic=self.topic, value=msg, key=self.kafka_msg_key)
+            list_fut_record_metadata = [
```
Comment (nkchern-avn) [inline journalpump/senders/kafka.py:113]: small: `list_fut_record_metadata` not a super descriptive name. Could it be `future_results` or smth. what this list contaons?

Comment (nkchern-avn) [reply journalpump/senders/kafka.py:113]: also this part: `... for msg in messages` kinda hints that they are messages...

File: journalpump/senders/kafka.py [hunk 2]
```diff
@@ -110,9 +110,14 @@ def send_messages(self, *, messages, cursor):
...
+                self.kafka_producer.send(topic=self.topic, value=msg, key=self.kafka_msg_key) for msg in messages
+            ]
             self.kafka_producer.flush()
+            for fut_record_metadata in list_fut_record_metadata:
```
Comment (nkchern-avn) [inline journalpump/senders/kafka.py:117]: or just `record_metadata`?

**Comment by nkchern-avn** [comment]
Would it be possible to have a test for this?

File: journalpump/senders/kafka.py (hunk 1 above)
Comment (mts-aiven) [reply journalpump/senders/kafka.py:113]: List of futures that resolve to record metadata objects.. but yep, could use result_futures as we do not actually care about the success values.

File: journalpump/senders/kafka.py (hunk 2 above)
Comment (mts-aiven) [reply journalpump/senders/kafka.py]: naming the loop variable `result_future` and the list `result_futures`

**Review by ivanyu — APPROVED** [review]

**Pull request merged**